File: modules/wem-api/src/test/java/com/enonic/wem/api/content/relationship/RelationshipTest.java
package com.enonic.wem.api.content.relationship;


import org.joda.time.DateTime;
import org.junit.Test;

import com.enonic.wem.api.account.UserKey;
import com.enonic.wem.api.content.MockContentId;
import com.enonic.wem.api.content.data.EntryPath;
import com.enonic.wem.api.content.schema.relationship.QualifiedRelationshipTypeName;

import static junit.framework.Assert.assertEquals;

public class RelationshipTest
{
    @Test
    public void build()
    {
        // setup
        final Relationship.Builder relationBuilder = Relationship.newRelationship();
        relationBuilder.fromContent( MockContentId.from( "a" ) );
        relationBuilder.toContent( MockContentId.from( "b" ) );
        relationBuilder.createdTime( DateTime.parse( "2012-01-01T12:00:00" ) );
        relationBuilder.creator( UserKey.from( "myStore:myUser" ) );
        relationBuilder.type( QualifiedRelationshipTypeName.from( "system:like" ) );
        relationBuilder.property( "stars", "4" );

        // exercise
        Relationship relationship = relationBuilder.build();

        // verify
        assertEquals( "a", relationship.getFromContent().toString() );
        assertEquals( "b", relationship.getToContent().toString() );
        assertEquals( "myUser", relationship.getCreator().getLocalName() );
        assertEquals( DateTime.parse( "2012-01-01T12:00:00" ), relationship.getCreatedTime() );
        assertEquals( "like", relationship.getType().getLocalName() );
        assertEquals( "4", relationship.getProperty( "stars" ) );
    }

    @Test
    public void getKey_having_relationship_with_mangingData()
    {
        // setup
        final Relationship.Builder relationBuilder = Relationship.newRelationship();
        relationBuilder.fromContent( MockContentId.from( "a" ) );
        relationBuilder.toContent( MockContentId.from( "b" ) );
        relationBuilder.type( QualifiedRelationshipTypeName.from( "system:like" ) );
        relationBuilder.managed( EntryPath.from( "myData" ) );

        // exercise
        Relationship relationship = relationBuilder.build();

        // verify
        assertEquals( "system:like", relationship.getKey().getType().toString() );
        assertEquals( "a", relationship.getKey().getFromContent().toString() );
        assertEquals( "b", relationship.getKey().getToContent().toString() );
        assertEquals( "myData", relationship.getKey().getManagingData().toString() );
    }

    @Test
    public void getKey_having_relationship_without_mangingData()
    {
        // setup
        final Relationship.Builder relationBuilder = Relationship.newRelationship();
        relationBuilder.fromContent( MockContentId.from( "a" ) );
        relationBuilder.toContent( MockContentId.from( "b" ) );
        relationBuilder.type( QualifiedRelationshipTypeName.from( "system:like" ) );

        // exercise
        Relationship relationship = relationBuilder.build();

        // verify
        assertEquals( "system:like", relationship.getKey().getType().toString() );
        assertEquals( "a", relationship.getKey().getFromContent().toString() );
        assertEquals( "b", relationship.getKey().getToContent().toString() );
        assertEquals( null, relationship.getKey().getManagingData() );
    }

}

File: modules/wem-webapp/src/test/webapp/admin2/api/js/relationship/RelationshipTest.ts
///<reference path='../TestCase.d.ts' />
///<reference path='../../../../../../main/webapp/admin2/api/js/relationship/module.ts' />
///<reference path='../../../../../../main/webapp/admin2/api/js/relationship/json/module.ts' />

TestCase("Relationship", {

    "test relationship": function () {

        var json:api_relationship_json.RelationshipJson = {
            id: 'test-relationship',
            type: 'relationship',
            fromContent: 'fromContent',
            toContent: 'toContent',
            managingData: 'data',
            properties: {},
            editable: true,
            deletable: false,

            modifiedTime: '02/02/2013',
            createdTime: '01/01/2013',
            creator: 'creator',
            modifier: 'modifier'
        };

        var relationship:api_relationship.Relationship = new api_relationship.Relationship(json);

        assertEquals(relationship.getType(), 'relationship');
        assertEquals(relationship.getCreator(), 'creator');
        assertEquals(relationship.getModifier(), 'modifier');
        assertEquals(relationship.getCreatedTime(), new Date('01/01/2013'));
        assertEquals(relationship.getModifiedTime(), new Date('02/02/2013'));
        assertEquals(relationship.getId(), 'test-relationship');
        assertEquals(relationship.getProperties(), {});
        assertEquals(relationship.getFromContent(), 'fromContent');
        assertEquals(relationship.getToContent(), 'toContent');


    }
});